File: merchantdemo/src/main/java/com/example/merchantdemo/service/StoreService.java
package com.example.merchantdemo.service;

import com.example.merchantdemo.dto.StoreDto;
import com.example.merchantdemo.entity.Store;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface StoreService {
    ResponseEntity<Store> updateStore(StoreDto theStore);

    ResponseEntity<List<Store>> findAllStores();

    ResponseEntity<Store> getStoreById(int theStoreNumber);

    ResponseEntity<List<Store>> getStoresByUserId(int userId);
}

File: merchantFrontEnd/src/Service/StoreService.js
import axios from "axios";

const STORE_API_BASE_URL = "http://localhost:8080/store";
class StoreService {
  getStoresByUserId(userId) {
    return axios.get(STORE_API_BASE_URL + "/all/" + userId);
  }

  getStoreById(theStoreNumber) {
    return axios.get(STORE_API_BASE_URL + "/" + theStoreNumber);
  }

  updateStore(theStore) {
    return axios.put(STORE_API_BASE_URL + "/" , theStore);
  }
}

export default new StoreService();
